src/MatrixClientPeg.ts

```diff
@@ -49,6 +49,13 @@ export interface IMatrixClientCreds {
     freshLogin?: boolean;
 }
 
+/**
+ * Holds the global instance of the `MatrixClient` to use across the codebase
+ * (singleton). Looking for an `MatrixClient`? Just look for the
```

Comment:
"singleton" isn't accurate for a MatrixClientPeg because the underlying/stored value can and does change throughout the application lifecycle. Thus, it holds the "current instance" rather than "global instance".

Reply:
I changed this to "current" and "pretty much a singleton". I'm not sure about the strict attributes of singletons but the goal is to give quick context which "singleton" conveys to me.

Reply:
[Singletons](https://en.wikipedia.org/wiki/Singleton_pattern) are generally expected to only have a single value through the lifetime of the application, though this can be relaxed to a lazily-created value if needed (such as in the case of many of the `instance` values on our stores)

Reply:
@turt2live To be clear, I see your point but think it's a nice way to describe what's pretty much going on. Language suggestions welcome to change this.

Reply:
moved to other thread